core/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -223,19 +224,21 @@ private void ensureNoPre019State() throws Exception {
      * index metadata with new settings.
      */
     private void pre20Upgrade() throws Exception {
-        MetaData metaData = loadMetaState();
-        List<IndexMetaData> updateIndexMetaData = new ArrayList<>();
-        for (IndexMetaData indexMetaData : metaData) {
+        List<PersistedIndexMetaData> metaData = metaStateService.loadIndicesStatesNoFolderNameValidation();
```

Comment:
The `loadMetaState` method does not load `IndexStateMetaData` (the persisted class) but rather just `IndexMetaData`, which makes sense because it is used to load the `MetaData` object for the cluster state.  I could have used `loadIndicesStates`, but this method assumes that the index folder name change upgrade has executed that @areek implemented (it throws an exception if the folder path != index uuid).  I needed a method that skips this check as this code is executed before the index folder upgrade.